app/http/ZipFilesResponse.php

```diff
@@ -54,9 +49,18 @@ private function compress()
             throw new ApiException("Archive could not be created");
         }
 
-        foreach ($this->files as $localPath => $name) {
-            if ($zip->addFile($localPath, $name) !== true) {
-                throw new ApiException("Error while adding file to archive");
+        foreach ($this->files as $name => $file) {
+            if ($file instanceof IImmutableFile) {
+                try {
+                    $file->addToZip($zip, $name);
+                } catch (FileStorageException $e) {
+                    throw new ApiException("Error while adding file to archive");
+                }
+            } else {
```

Comment:
Is this else really needed, is there situation where this would be used?

Reply:
You are right, it is not used anywhere. But I was thinking a bit ahead trying to make it reusable...